### PR TITLE
De-circularize Titles and Axes

### PR DESCRIPTION
**Pull request by etpinard**
@mdtusz 

another circular dependency pattern bites the dust :dash:

### CODE REVIEW

File: src/components/titles/index.js [hunk 1]
```diff
@@ -239,16 +242,16 @@ Titles.draw = function(gd, title) {
                 // iterate over a set of elements (avoid.selection)
                 // to avoid collisions with
                 avoid.selection.each(function(){
-                    var avoidbb = Plotly.Drawing.bBox(this);
+                    var avoidbb = Drawing.bBox(this);
 
-                    if(Plotly.Lib.bBoxIntersect(titlebb,avoidbb,pad)) {
+                    if(Lib.bBoxIntersect(titlebb,avoidbb,pad)) {
```
Comment (mdtusz) [inline src/components/titles/index.js]: Let's space out these args.

Comment (etpinard) [inline src/components/titles/index.js]: good :eyes:

File: src/components/colorbar/draw.js [hunk 2]
```diff
@@ -253,10 +254,10 @@ module.exports = function draw(gd, id) {
 
         cbAxisOut._axislayer = container.select('.cbaxis');
         var titleHeight = 0;
-        if(['top','bottom'].indexOf(opts.titleside)!==-1) {
+        if(['top', 'bottom'].indexOf(opts.titleside) !==- 1) {
```
Comment (mdtusz) [inline src/components/colorbar/draw.js]: Looks like the space was a bit too eager there.

Should be `if(['top', 'bottom'].indexOf(opts.titleside) !== -1) {`

**Comment by mdtusz** [comment]
Looks ok after those two minor :cow2:'s.

:dancer: once they're remedied!

**Pull request merged**